pkg/pillar/cmd/domainmgr/metric.go

```diff
@@ -92,10 +98,20 @@ func getAndPublishMetrics(ctx *domainContext, hyper hypervisor.Hypervisor) {
 				continue
 			}
 			dm.Activated = status.Activated
+			// Scale the CPU nanoseconds based on the number of VCpus
+			if status.VCpus != 0 {
+				dm.CPUTotalNs /= uint64(status.VCpus)
+				dm.CPUScaled = uint32(status.VCpus)
+			}
+		} else if dm.UUIDandVersion.UUID == nilUUID {
+			// Scale Xen Dom0 based CPUs seen by hypervisor
+			dm.CPUTotalNs /= uint64(hm.Ncpus)
```

Comment:
Tests hit division by zero on this line. Looks like some problems with parsing inside GetHostCPUMem() for xen. Probably we must return error there in case of no information for particular lines we try to parse from output of xl info.